octue/cloud/pub_sub/service.py

```diff
@@ -187,23 +188,43 @@ def ask(self, service_id, input_values, input_manifest=None):
     def wait_for_answer(self, subscription, timeout=30):
         """Wait for an answer to a question on the given subscription, deleting the subscription and its topic once
         the answer is received.
+
+        :param octue.cloud.pub_sub.subscription.Subscription subscription: the subscription for the question's answer
+        :param float timeout: how long to wait for an answer before raising a TimeoutError
+        :raise TimeoutError: if the timeout is exceeded
+        :return dict: dictionary containing the keys "output_values" and "output_manifest"
         """
-        pull_response = self.subscriber.pull(
-            request={"subscription": subscription.path, "max_messages": 1},
-            timeout=timeout,
-            retry=create_custom_retry(timeout),
-        )
+        start_time = time.perf_counter()
+        no_message = True
+        attempt = 1
+
+        with self.subscriber:
+            while no_message:
+                logger.debug("Pulling messages from Google Pub/Sub: attempt %d.", attempt)
+
+                pull_response = self.subscriber.pull(
+                    request={"subscription": subscription.path, "max_messages": 1},
+                    retry=create_custom_retry(timeout),
```

Comment:
I presume that we can't handle this in the custom_retry; because that'll only get called in the event of an actual error?

Reply:
I'm not sure. Did you ever see an error being raised by Google? I don't think I did, the pull response just returned with no messages. Maybe an error is raised under the hood?

I could try a `Retry` with all possible exceptions and see if that covers it?

Reply:
No, I never saw anything raised by google

Reply:
Are we ok to keep this as is then as it seems to work?

Reply:
Yes I think so. We’ve stepped it forward and probably solved the actual issue, right?